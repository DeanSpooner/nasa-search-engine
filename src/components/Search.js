import React, { useState } from "react";
import getImages from "../requests/getImages";
import PropTypes from "prop-types";
import "../styles/Search.css";

const Search = ({ setSearchResults }) => {
  const [value, setValue] = useState(0);

  const handleSubmit = async (event) => {
    event.preventDefault();
    setSearchResults(await getImages(value));
  };

  return (
    <>
      <form className="search__form" onSubmit={handleSubmit}>
        <input
          type="text"
          className="search__input"
          data-testid="search__input"
          onChange={(e) => setValue(e.target.value)}
        />
        <button
          className="search__button"
          data-testid="search__button"
          type="submit"
        >
          Search
        </button>
      </form>
    </>
  );
};

Search.propTypes = {
  setSearchResults: PropTypes.func.isRequired,
};

export default Search;
