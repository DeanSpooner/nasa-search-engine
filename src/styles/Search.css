form {
  display: flex;
  flex-direction: row;
  justify-content: center;
}

.search__input {
  width: 40%;
}
