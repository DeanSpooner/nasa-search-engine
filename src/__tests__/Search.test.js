import React from "react";
import { render } from "@testing-library/react";
import Search from "../components/Search";

describe("Search", () => {
  const validProps = {
    setSearchResults: () => {},
  };

  const { asFragment } = render(<Search />);

  it("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders input correctly", () => {
    const { getByTestId } = render(
      <Search setSearchResults={validProps.setSearchResults} />
    );
    const input = getByTestId("search__input");
    expect(input).toBeTruthy();
  });

  it("renders button correctly", () => {
    const { getByTestId } = render(
      <Search setSearchResults={validProps.setSearchResults} />
    );
    const button = getByTestId("search__button");
    expect(button).toBeTruthy();
  });
});
