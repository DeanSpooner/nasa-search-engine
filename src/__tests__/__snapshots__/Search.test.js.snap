// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <form
    class="search__form"
  >
    <input
      class="search__input"
      data-testid="search__input"
      type="text"
    />
    <button
      class="search__button"
      data-testid="search__button"
      type="submit"
    >
      Search
    </button>
  </form>
</DocumentFragment>
`;
